src/modules/storeTask.js

```diff
@@ -0,0 +1,36 @@
+const getTask = () => {
+  let tasks;
+  if (localStorage.getItem('data') === null) {
+    tasks = [];
+  } else {
+    tasks = JSON.parse(localStorage.getItem('data'));
+  }
+  return tasks;
+};
+
+const addTaskItem = (task) => {
+  const tasks = getTask();
+  tasks.push(task);
+  localStorage.setItem('data', JSON.stringify(tasks));
+};
+
+const deleteTask = (index) => {
+  const tasks = getTask();
+  const deletedTasks = tasks.filter((task) => task.index !== index);
+  deletedTasks.forEach((task, i) => {
+    task.index = i;
+  });
```

Comment:
Very well done by implementing the delete functionality but when updating indexes you starts from 0 which is not the requested output we need the first item to have index 1 and so on you can always asign i+1 

[OPTIONAL] this can be done in one loop for better performance 
`const deletedTasks = tasks.filter((task, i) => {
    let found = false;
    if(task.index == index){
      found = true
    }else{
      return  found ? {...task, index: i - 1} : task
    }
    });`